services/filter/src/filter/routes/filter.py

```diff
@@ -0,0 +1,240 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+import logging
+import os.path
+from typing import Optional
+
+import duckdb
+import pyarrow as pa
+import pyarrow.parquet as pq
+from datasets import Features, Value
+from libapi.authentication import auth_check
+from libapi.exceptions import (
+    ApiError,
+    InvalidParameterError,
+    MissingRequiredParameterError,
+)
+from libapi.utils import (
+    Endpoint,
+    are_valid_parameters,
+    get_json_api_error_response,
+    get_json_ok_response,
+    to_rows_list,
+)
+from libcommon.exceptions import UnexpectedError
+from libcommon.parquet_utils import ParquetFileMetadataItem
+from libcommon.processing_graph import ProcessingGraph
+from libcommon.prometheus import StepProfiler
+from libcommon.simple_cache import get_previous_step_or_raise
+from libcommon.storage import StrPath
+from libcommon.utils import PaginatedResponse
+from libcommon.viewer_utils.features import (
+    get_supported_unsupported_columns,
+    to_features_list,
+)
+from starlette.requests import Request
+from starlette.responses import Response
+
+MAX_ROWS = 100
```

Comment:
nit: factor this constant and other parts of the file that are shared with /rows and/or /search.

Reply:
Done.